### PR TITLE
indexer: Implement package indexer for maven

### PR DESCRIPTION
**Pull request by arajkumar**
Maven is a package repository for Java.

The mvn tool which packs the maven projects usually adds metadata fields(dependency name, version) into `META-INF/maven/<groupId>/<artifactId>` which could be found on every jar file.

There are two files with in every maven jar which would give us the necessary information like package|version,

```
META-INF/maven///pom.properties
META-INF/maven///pom.xml
```
>[1] The pom.xml and pom.properties files are packaged up in the JAR so that each artifact produced by Maven is self-describing and also allows you to utilize the metadata in your own application if the need arises. One simple use might be to retrieve the version of your application. Operating on the POM file would require you to use some Maven utilities but the properties can be utilized using the standard Java API and look like the following:

```
version=1.0-SNAPSHOT
groupId=com.mycompany.app
artifactId=my-app
```

[1] https://maven.apache.org/guides/getting-started/#how-do-i-add-resources-to-my-jar

`github.com/aquasecurity/go-dep-parser/pkg/jar` has been used to parse jar/war/ear files.

Fixes https://github.com/quay/claircore/issues/236

Signed-off-by: Arunprasad Rajkumar <ar.arunprasad@gmail.com>

### CODE REVIEW

**Comment by arajkumar** [comment]
Trivy is using Github Advisory as a data source: https://github.com/advisories?query=ecosystem%3Amaven

Gitlab also has public security advisory database for maven: https://gitlab.com/gitlab-org/security-products/gemnasium-db/-/tree/master/maven

File: java/packagescanner.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	libs, err := jar.Parse(r, jar.WithFilePath(name))
+	if err != nil {
+		return nil, err
+	}
+	packages := make([]*claircore.Package, len(libs))
+	for i, l := range libs {
+		packages[i] = &claircore.Package{
+			Name:           l.Name,
+			Version:        l.Version,
+			PackageDB:      "maven:" + name,
```
Comment (ldelossa) [inline java/packagescanner.go:105]: Referencing the python package scanner, is there any path fixing that needs to occur here, similar to:
https://github.com//ldelossa/claircore/blob/b165424edcf12301661573730041058b505d716c/python/packagescanner.go#L81 ?

**Review by ldelossa — CHANGES_REQUESTED** [review]

File: java/packagescanner.go (hunk 1 above)
Comment (arajkumar) [reply java/packagescanner.go:105]: It is not necessary here, we just check at the extension without any path.

Comment (ldelossa) [reply java/packagescanner.go:105]: Does this result in the PackageDB field showing where the java package exists in the file system?

Comment (arajkumar) [reply java/packagescanner.go:105]: Yeah, that is a valid question. I assumed it should have full file path.

However the python implementation seem to be recording the directory which holds the package in filesystem. Let me fix it.

Comment (ldelossa) [reply java/packagescanner.go:105]: Yes, this field should be the file system location where the package was found. It's used for uniqueness so we can find the same package across multiple file system locations.

**Review by ldelossa — APPROVED** [review]

**Pull request merged**